tools/Vrops.py

```diff
@@ -105,21 +105,24 @@ def get_resources(self, target: str,
                       adapterkind: str,
                       resourcekinds: list,  # Array of resource kind keys
                       uuids: list,  # Array of parent uuids
-                      query_specs: dict,  # Dict of query specifications
+                      query_specs: dict,  # Dict of resource query specifications
                       ) -> (list, int):
         if not uuids:
             logger.debug(f'No parent resources for {resourcekinds} from {target}')
             return [], 200
-        logger.debug(f'Getting {resourcekinds} from {target}, query specs: {query_specs}')
+        logger.debug(f'Getting {resourcekinds} from {target}')
         url = "https://" + target + "/suite-api/api/resources/bulk/relationships"
+
+        # If there is a config for the resourcekind in question, use it, otherwise give back the default.
+        q_spec = query_specs.get(resourcekinds[0]) if query_specs.get(resourcekinds[0]) else query_specs.get('default')
```

Comment:
What happens if multiple resourcekinds are getting passed? Wouldn't we just look at index 0?

Reply:
Only one query specification can be used per query, even if there are several different ones for the resourcekinds. So the query specifications of the first mentioned resourcekind are always used.

Reply:
Wouldn't that mean, that the config possibly wouldn't fit for the other resourcekinds if multiple are passed?

Reply:
Yes. The only thing we can do is to make only one request per resourcekind at a time. But this again increases the overhead on the API.

Reply:
But I also welcome your suggestion on this.

Reply:
Well if we have different settings specified in `inventory_config` we must not ignore it and take the first one only. That would break every expected behaviour.

We should sort it out here, when we have multiple resourcekinds defined in `inventory_config` we've got to make single calls. The rest should be grouped und queried with the default query specs.

Make single resourcekind calls for the resourcekinds defined explicitly in `inventory_config` and for the rest we do a default call. To avoid duplication here we should move the actual `payload`-`request` handling to a dedicated function. like `get_resources_request` and feed in the seperated query specs, resourcekinds which are fitting (multiple for default, single resourcekinds for specifics).